src/java/org/apache/cassandra/config/DatabaseDescriptor.java

```diff
@@ -4763,4 +4756,159 @@ public static OptionalDouble getSeverityDuringDecommission()
+    {
+        try
+        {
+            for (Property property : ofNullable(confValueAccessors).orElse(Collections.emptyMap()).values())
+                visitor.visit(property.getName(), property.getType(), property.getAnnotation(Mutable.class) == null);
+        }
+        catch (Throwable t)
+        {
+            throw handler.apply(t);
+        }
+    }
+
+    public static <T> ListenableProperty.Remover addBeforeChangePropertyListener(String name, Class<T> clazz, BiConsumer<T, T> consumer)
+    {
+        return addPropertyListener(name, clazz, p -> p.addBeforeHandler(ListenableProperty.Handler.consume(consumer)));
+    }
+
+    public static <T> ListenableProperty.Remover addAfterChangePropertyListener(String name, Class<T> clazz, BiConsumer<T, T> consumer)
+    {
+        return addPropertyListener(name, clazz, p -> p.addAfterHandler(ListenableProperty.Handler.consume(consumer)));
+    }
+
+    @SuppressWarnings("unchecked")
+    private static <T> ListenableProperty.Remover addPropertyListener(String name,
+                                                                      Class<T> clazz,
+                                                                      Function<ListenableProperty<Config, T>, ListenableProperty.Remover> adder)
+    {
+        Property property = ofNullable(confValueAccessors.get(name))
+                            .map(p -> {
+                                // The case when replacement property is used and match the property's name.
+                                if (p instanceof Replacement.ReplacementProperty)
+                                    return ((Replacement.ReplacementProperty) p).delegate();
+                                else
+                                    return p;
+                            })
+                            .orElseThrow(() -> new PropertyNotFoundException(name));
+        if (!(property instanceof ListenableProperty))
+            throw new ConfigurationException(String.format("Property '%s' is not listenable.", name));
+
+        if (clazz.equals(property.getType()))
+            return adder.apply((ListenableProperty<Config, T>) property);
+        else
+            throw new ConfigurationException(String.format("Listener type '%s' doesn't match the property '%s' type '%s'.",
+                                                           clazz.getSimpleName(), name, clazz.getSimpleName()));
+    }
+
+    public static Object getProperty(String name)
+    {
+        return ofNullable(confValueAccessors.get(name)).orElseThrow(() -> new PropertyNotFoundException(name))
+                                                       .get(conf);
+    }
+
+    public static String getStringProperty(String name)
+    {
+        Property property = ofNullable(confValueAccessors.get(name))
+                            .orElseThrow(() -> new PropertyNotFoundException(name));
```

Comment:
please refactor so `getProperty` has the same logic, can do something simple like

```
private static Property getProp(String name)
{
  Property prop = confValueAccessor.get(name);
  if (prop == null) new PropertyNotFoundException(name);
  return prop;
}
```

Then in both methods you call `getProp(name)` and avoid duplicate check code